controlplane/kubeadm/api/v1alpha4/kubeadm_control_plane_webhook.go

```diff
@@ -64,6 +65,26 @@ func (in *KubeadmControlPlane) Default() {
 	if !strings.HasPrefix(in.Spec.Version, "v") {
 		in.Spec.Version = "v" + in.Spec.Version
 	}
+
+	// Enforce RollingUpdate strategy and default MaxSurge and MaxUnavailable if not set.
+	if in.Spec.RolloutStrategy != nil {
+		if in.Spec.RolloutStrategy.Type != RollingUpdateStrategyType {
+			in.Spec.RolloutStrategy.Type = RollingUpdateStrategyType
+		}
+
+		if in.Spec.RolloutStrategy.RollingUpdate == nil {
+			in.Spec.RolloutStrategy.RollingUpdate = &RollingUpdate{}
+		}
+
+		if in.Spec.RolloutStrategy.RollingUpdate.MaxSurge == nil {
+			ios1 := intstr.FromInt(1)
```

Comment:
what if `MaxUnavailable` is set to 1 but `MaxSurge` isn't set?

Reply:
(and vice versa)

Reply:
@CecileRobertMichon good point. We definitely need to check if the user has only set MaxSurge or MaxUnavailable. In that case, we need to check the existing value and set the missing MaxSurge or MaxUnavailable value accordingly. I'll fix this.